### ISSUE TITLE
Redis session data serialized twice

### ISSUE THREAD

**Issue by redigaffi** [laravel/framework#27072]
- Laravel Version: 5.5.*
- PHP Version: 7
- Database Driver & Version: Not needed

### Description:
I'm using as session storage the Redis service, all works fine, but having a look manually in the values stored in the Redis server it seems that Laravel serializes twice the data. 

**dd() before saving data in Redis:**
`a:4:{s:6:"_token";s:40:"nSh5GbbZplACOiAtPz13QVWqnANf8KEPhph4LDgX";s:11:"play_status";s:6:"queued";s:9:"_previous";a:1:{s:3:"url";s:27:"http://localhost:8081/login";}s:6:"_flash";a:2:{s:3:"old";a:0:{}s:3:"new";a:0:{}}}`

**Retrieving values manually in the Redis server:**
`"s:189:\"a:3:{s:6:\"_token\";s:40:\"ztSydX6zcRITF7zRr9bkAYjtdzLU14fun4edacAf\";s:6:\"_flash\";a:2:{s:3:\"old\";a:0:{}s:3:\"new\";a:0:{}}s:9:\"_previous\";a:1:{s:3:\"url\";s:31:\"http://localhost:8081/register1\";}}\";"
`

As you can see its serialized twice, not sure if this is intended, but I need to unserialize this session data in node.js and I'm having trouble because of this issue.

### Steps To Reproduce:
Just use Redis handler for sessions and manually retrieve session data in the Redis server.

**Comment by thecrypticace** [comment]
This is because the session handler for Redis delegates to the cache layer to handle writing data to Redis.

The session store serializes data on save regardless of the handler: https://github.com/laravel/framework/blob/78505345f2a34b865a980cefbd103d8eb839eedf/src/Illuminate/Session/Store.php#L129

And the underlying cache-based handler for Redis also serializes data: https://github.com/laravel/framework/blob/78505345f2a34b865a980cefbd103d8eb839eedf/src/Illuminate/Cache/RedisStore.php#L93

So, whether or not it is intended it's "fine" because the appropriate unserialize operations also get called when reading data. This seems rather inefficient to me but I doubt changing this would actually result in any meaningful perf improvements. As for reading the data from NodeJS you'll just have to know that the data is double serialized and will have to decode accordingly.

Alternatively, you could write your own session handler that communicates with Redis directly and serializes only once _or_ even writes to Redis using JSON instead of PHP's serialization format which I imagine would make parsing the data simpler.

**Comment by redigaffi** [comment]
Actually that's what I did, I wrote a custom Session Handler that stores the data in redis as JSON format, the problem with that is if you have a object, like ViewBagError, it get's lost because JSON don't store the meta-data of classes. I was stuck days because he validation errors wasn't showing, until I realized that the ViewBagError was getting lost when encoding/decoding in JSON format. I guess the best option will be writing the cusotm handler to store it serialized once in php and try to unserialize it in Node.JS.
